### DISCUSSION TITLE
Add support for indeterminate checkbox state like [-]

### DISCUSSION THREAD

**Discussion by floer32** [community/community#19199] [Projects and Issues]
This is a suggestion about GitHub-Flavored Markdown but I want it specifically in Issues/Pull Requests. (It doesn't matter to me if it would be supported in READMEs etc., although it'd probably make sense to support it the same everywhere, if implemented.)

It would be great to be able to use `[-]` to mark a checkbox as partially checked / indeterminate state. This would be useful in nested checklists.

A bit more specifically, here's an example from an open issue in the `marktext` repository (https://github.com/marktext/marktext/issues/2819):

> Input:
> 
> ```markdown
> - [x] Item 1
> - [-] Item 2
> - [ ] Item 3
> ```
> 
> Output:
> 
> ![image](https://user-images.githubusercontent.com/2420688/175111582-5ea66d6c-af16-4897-a6b3-31548eabfd04.png)

Example usage to provide context:

```markdown
- [x] Todo
    - [x] Sub-item
    - [x] Sub-item 
- [-] Another todo
    - [x] Sub-item [completed]
    - [-] Sub-item[^1]
- [x] Another todo
- [ ] Another todo

[^1]: Maybe this item has been obviated by PR #123
```

I recognize that [it's a feature of the JavaScript DOM, not HTML itself](https://css-tricks.com/indeterminate-checkboxes/) but there are already other features of checklists in GitHub Issues and Pull Requests, that are "rich" functionality without an HTML equivalent: like the ability to drag and drop checkbox items to re-sort them, click an icon to "convert to issue," and how checkbox state is actually saved on clicking.

**Comment by AlanSl** [comment]
This would be great and it's really worth highlighting how useful it'd be specifically for GitHub issues, for flat list use cases similar to your "obviated by PR #123" example. 

There are lots of scenarios in Github issue management where a subtask is no longer "to do", but is not done in the context of the current issue or PR either, where marking the item off as `[-]` with something other than a tick would be really useful, e.g.:

 - Subtasks split out into separate issues
 - Subtasks already solved by other work
 - Redundant subtasks that proved to be unnecessary on closer inspection
 - "wontfix" subtasks that proved to be unfeasible or undesirable to fix

Having three states on checklists would make it really easy to quickly see what's done, what's still to do, and what was considered but not done in the current issue. I often do this and currently use ugly workarounds like converting the checkbox to a regular bullet so it doesn't give a misleading impression.

**Comment by addisonklinke** [comment]
It would be nice for partially checked boxes to have grayed-out text to make the difference more obvious. Just the difference between check and slash is a little too subtle IMO

**Comment by dustinbowers** [comment]
I'm looking for this feature, and I'm surprised it doesn't exist already. I think it's fair to reason a tasklist item can be "mostly completed" without warranting a full checkmark. It'd be a useful level of distinction

**Comment by roke-julian-lockhart** [comment]
> [`#discussion-4165815`](https://github.com/orgs/community/discussions/19199#discussion-4165815)

@floer32, note that this functionality is also being tracked at GitLab's issue tracker, at [`gitlab.com/gitlab-org/gitlab/-/issues/229189`](https://gitlab.com/gitlab-org/gitlab/-/issues/229189).

**Comment by benhodgson87** [comment]
Would also find this really useful. We have a readiness checklist in PR templates, and being able to mark a checkbox with a dash to signal "This is unnecessary for this change", rather than it appearing as "I haven't done this", would be very helpful.

**Comment by dlight** [comment]
Is this on Github's radar? There are a lot of Markdown files in the wild that have the half-done state `[-]`.

We just need Github-flavored Markdown to be extended with this new form of tasklist marker (besides `[ ]` and `[x]`), and parsers will eventually pick this up.

Per the link above, the decided it was "prospective" 5 months ago ("Indicates that issues describes a bug, feature or maintenance effort that is in line with GitLab strategy and GitLab team members can take it on in the next 18 months.").

Note that Gitlab also supports crossing out a task in a way that disables its editbox, [using the marker `[~]`](https://gitlab.com/gitlab-org/gitlab/-/merge_requests/85982). If Github will ever revise its tasklist syntax for Markdown, it could add support for both.
